content/blog/2017/2017-02-07-declarative-maven-project.adoc

```diff
@@ -235,17 +238,55 @@ pipeline {
                 sh 'mvn -Dmaven.test.failure.ignore=true install' // <1>
             }
             post {
-                always {
+                success {
                     junit 'target/surefire-reports/**/*.xml' // <2>
                 }
             }
         }
     }
 }
-// Scripted Not Defined //
+// Scripted
+node {
+    checkout scm
+
+    String jdktool = tool name: "jdk8", type: 'hudson.model.JDK'
+    def mvnHome = tool name: 'mvn'
+
+    /* Set JAVA_HOME, and special PATH variables. */
+    List javaEnv = [
+        "PATH+MVN=${jdktool}/bin:${mvnHome}/bin",
+        "M2_HOME=${mvnHome}",
+        "JAVA_HOME=${jdktool}"
+    ]
+
+    withEnv(javaEnv) {
+    stage ('Initialize') {
+        sh '''
+            echo "PATH = ${PATH}"
+            echo "M2_HOME = ${M2_HOME}"
+        '''
+    }
+    stage ('Build') {
+        try {
+            sh 'mvn -Dmaven.test.failure.ignore=true install'
+        } catch (e) {
+            currentBuild.result = 'FAILURE'
+        }
+    }
+    stage ('Post') {
+        if (currentBuild.result == null || currentBuild.result == 'SUCCESS') {
+            junit 'target/surefire-reports/**/*.xml'  // <2>
+        }
```

Comment:
As in the previous comment: if "Build" fails, then we won't reach this stage, so there's no need for the wrapping `if` statement.